### PR TITLE
Clarify collection paths in docs

### LINKED ISSUE #1231: Clearer documentation of how to run playbook with modified clone
##### SUMMARY

I was trying to write a PR, and I struggled to get Ansible to use my modified clone of this repo, instead of the stable pre-installed version.

I believe the documentation is insufficient. In particular `COLLECTIONS_PATHS` should **not** point to the cloned repo, nor the parent of the cloned repo. This is *extremely* counter intuitive, and so should be spelled out.

##### ISSUE TYPE
- Documentation Report

##### COMPONENT NAME
documentation

##### ANSIBLE VERSION

N/A, `master` branch

# Issues

1. Start on [the Ansible Developer Guide](https://docs.ansible.com/ansible/latest/dev_guide/index.html)
1. Click on "I want to contribute to an Ansible-maintained collection.", which takes you [here](https://docs.ansible.com/ansible/latest/community/contributing_maintained_collections.html#contributing-maintained-collections)
1. Click on "Developer Guide" for "community.general
1. Now we're back to the first page. We just went in a big circle
1. Ctrl-F for "collection" and scroll down. Click on [Developing Collections](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections.html)
1. Click on ["Trying Collections Locally"](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections.html#trying-collections-locally). That sounds like it has what we need!
1. Nope. The section named "Trying Collections Locally" does not explain how to try collections locally. It explains how to install from a .tar ball. So am I supposed to tar and re-install every time I change one line in the repo and want to build?
1. Ok I guess the Ansible Developer guide won't help us. Let's check the README of this repo.

> If you want to develop new content for this collection or improve what is already here, the easiest way to work on the collection is to clone it into one of the configured COLLECTIONS_PATH, and work on it there.

Ok, so I clone into `~/dev/community.general`, and set `COLLECTIONS_PATH` to `~/dev`. 

Hmm, that doesn't work. Let's check [the documentation linked to by the README](https://docs.ansible.com/ansible/latest/reference_appendices/config.html#collections-paths).

> Colon separated paths in which Ansible will search for collections content.

Ok, yes. We want Ansible to search for collections in the direct parent directory of the clone. So we've configured `COLLECTIONS_PATH` as per the documentation. Why won't it work?

Apparently I'm supposed to clone into some random directory *four layers deep*! (`~/collections/ansible_collections/community/general/`) That's not obvious. 

# Solution

The README for this repo should contain the crucial parts of [this documentation](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections.html#contributing-to-collections). 

i.e. 

```
mkdir -p collections/ansible_collections/community/
git clone git@github.com:ansible-collections/community.general.git collections/ansible_collections/community/general
export COLLECTIONS_PATH=$(pwd)/collections:$COLLECTIONS_PATH
```

Now you might argue that this is not necessary because this is explained in Ansible's documentation. But as I explained above, it's not. The documentation has circular links, misleading headlines, and `COLLECTIONS_PATH` has a misleading explanation.

### PR DESCRIPTION
**Pull request by mdavis-xyz**
##### SUMMARY

Fixes [1231 from community.general](https://github.com/ansible-collections/community.general/issues/1231).

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME
documentation

### CODE REVIEW

**Comment by mdavis-xyz** [comment]
Note that I'm not familiar with `rst`. So please double check the `:ref:` part.

**Comment by mdavis-xyz** [comment]
This is related to #72494

**Comment by mdavis-xyz** [comment]
The Shippable CI tests failed, but I'm having a hard time understanding the logs to see what went wrong.

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 1]
```diff
@@ -400,6 +400,8 @@ You should use one of the values configured in :ref:`COLLECTIONS_PATHS` for your
 expect to find collections when attempting to use them. If you don't specify a path value, ``ansible-galaxy collection install``
 installs the collection in the first path defined in :ref:`COLLECTIONS_PATHS`, which by default is ``~/.ansible/collections``.
 
+If you want to use a collection from a folder rather than a tarball, see :ref:`hacking_collections`.
```
Comment (felixfontein) [inline docs/docsite/rst/dev_guide/developing_collections.rst]:
```suggestion
If you want to use a collection directly out of a checked out git repository, see :ref:`hacking_collections`.
```

File: lib/ansible/config/base.yml [hunk 2]
```diff
@@ -243,7 +243,12 @@ COLLECTIONS_SCAN_SYS_PATH:
   - {key: collections_scan_sys_path, section: defaults}
 COLLECTIONS_PATHS:
   name: ordered list of root paths for loading installed Ansible collections content
-  description: Colon separated paths in which Ansible will search for collections content.
+  description: >
+    Colon separated paths in which Ansible will search for collections content.
+    Collections must be in nested *subdirectories*, not directly in these directories.
+    For example, if `COLLECTIONS_PATHS` includes `~/.ansible/collections`,
+    and you want to add `my.collection`, it must be saved as
+    `~/.ansible/collections/ansible_collections/my/collection`.
```
Comment (felixfontein) [inline lib/ansible/config/base.yml]:
```suggestion
    For example, if ``COLLECTIONS_PATHS`` includes ``~/.ansible/collections``,
    and you want to add ``my.collection`` to that directory, it must be saved as
    ``~/.ansible/collections/ansible_collections/my/collection``.
```

**Comment by mdavis-xyz** [comment]
@felixfontein I just accepted your two proposed changes.

**Review by felixfontein — APPROVED** [review]

**Comment by acozine** [comment]
This looks like a useful clarification. Thanks @mdavis-xyz and I'm glad the test suite finally passed.

**Pull request merged**